pkg/apis/core/v1beta1/helper/helper.go

```diff
@@ -782,3 +699,126 @@ func FindPrimaryDNSProvider(providers []gardencorev1beta1.DNSProvider) *gardenco
+		for _, p := range predicate {
+			if p == nil {
+				continue
+			}
+			semver, err := semver.NewVersion(v.Version)
+			if err != nil {
+				return gardencorev1beta1.ExpirableVersion{}, fmt.Errorf("error while parsing version '%s': %s", v.Version, err.Error())
+			}
+			shouldFilter, err := p(v, semver)
+			if err != nil {
+				return gardencorev1beta1.ExpirableVersion{}, fmt.Errorf("error while evaluation predicate: '%s'", err.Error())
+			}
+			if shouldFilter {
+				continue OUTER
+			}
+		}
+
+		filteredVersions[v.Version] = v
+	}
+	if len(filteredVersions) == 0 {
+		return gardencorev1beta1.ExpirableVersion{}, fmt.Errorf("no latest qualifying version could be determined")
+	}
+	latestVersion := sets.StringKeySet(filteredVersions).List()[len(filteredVersions)-1]
+	return filteredVersions[latestVersion], nil
+}
+
+// FilterForConsecutiveMinor returns a closure that evaluates whether a given version v has a consecutive minor version compared to the currentSemVerVersion
+// returns true if v has a consecutive minor version
+func FilterForSameMajorMinorVersion(currentSemVerVersion semver.Version) func(expirableVersion gardencorev1beta1.ExpirableVersion, version *semver.Version) (bool, error) {
+	return func(_ gardencorev1beta1.ExpirableVersion, v *semver.Version) (bool, error) {
+		isWithinRange, err := versionutils.CompareVersions(v.String(), "~", currentSemVerVersion.String())
+		if err != nil {
+			return true, err
+		}
+		return !isWithinRange, nil
+	}
+}
+
+// FilterForConsecutiveMinor returns a closure that evaluates whether a given version v has a consecutive minor version compared to the currentSemVerVersion
+// returns true if v has a consecutive minor version
+func FilterForConsecutiveMinor(currentSemVerVersion semver.Version) func(expirableVersion gardencorev1beta1.ExpirableVersion, version *semver.Version) (bool, error) {
+	return func(_ gardencorev1beta1.ExpirableVersion, v *semver.Version) (bool, error) {
+		isWithinRange, err := versionutils.CompareVersions(v.String(), "^", currentSemVerVersion.String())
+		if err != nil {
+			return true, err
+		}
+
+		if !isWithinRange {
+			return true, nil
+		}
+
+		hasIncorrectMinor := currentSemVerVersion.Minor()+1 != v.Minor()
+		return hasIncorrectMinor, nil
+	}
+}
+
+// FilterSameVersion returns a closure that evaluates whether a given version v is equal to the currentSemVerVersion
+// returns true it it is equal
+func FilterSameVersion(currentSemVerVersion semver.Version) func(expirableVersion gardencorev1beta1.ExpirableVersion, version *semver.Version) (bool, error) {
+	return func(_ gardencorev1beta1.ExpirableVersion, v *semver.Version) (bool, error) {
+		return v.Equal(&currentSemVerVersion), nil
+	}
+}
+
+// FilterExpiredVersionPredicate returns a closure that evaluates whether a given expirable version is expired
+// returns true it it is expired
+func FilterExpiredVersionPredicate() func(expirableVersion gardencorev1beta1.ExpirableVersion, version *semver.Version) (bool, error) {
```

Comment:
Can you rename to keep the predicate names consistent?
```suggestion
func FilterExpiredVersion() func(expirableVersion gardencorev1beta1.ExpirableVersion, version *semver.Version) (bool, error) {
```